packages/babel-parser/src/parse-error/module-errors.js

```diff
@@ -0,0 +1,14 @@
+// @flow
+
+import { ParseErrorCodes, toParseErrorCredentials } from "../parse-error";
+
+export default (_: typeof toParseErrorCredentials) => ({
```

Comment:
Nit: to reduce the boilerplate, we could export `type _ = typeof toParseErrorCredentials` from `parse-error.js`, and just to
```js
export default (_: _) => ({
```

Reply:
Yeah, this one is really unfortunate. If you look at the flow/typescript/placeholder/etc. errors, they don't need this at all, because they just call `toParseErrorClasses(_ => ...)`, and so get the function's type inferred for free. The standard errors can't do this because I want to export them *from* parse-error, and do to the infuriating way that import works (this wouldn't be a problem using `require`), I can't use `toParseErrorClasses` in these files (see me complaining about this here: https://twitter.com/tolmasky/status/1492204196019453952?s=21 ). I chose not to export `toParseErrorCredentials` for this reason, since its an ugly artifact of the Flow/ESM combination (it wouldn't be necessary with Typescript/ESM *or* with Flow/require), we don't need it for any of the plugin error definitions, and we can hopefully completely get rid of it when we switch to Typescript. But, happy to do this one if you want, just wanted to explain the reasoning (or rather, just vent about ESM and Flow ;) )